src/physics/model.h

```diff
@@ -26,24 +27,76 @@ class ModelBase
+    /// Convective flux terms additional to the baseline physics (including convective flux terms in additional PDEs of model)
     virtual std::array<dealii::Tensor<1,dim,real>,nstate> 
     convective_flux (
         const std::array<real,nstate> &conservative_soln) const = 0;
 
-    /// Dissipative flux terms additional to the baseline physics
+    /// Dissipative flux terms additional to the baseline physics (including dissipative flux terms in additional PDEs of model)
 	virtual std::array<dealii::Tensor<1,dim,real>,nstate> 
 	dissipative_flux (
     	const std::array<real,nstate> &conservative_soln,
     	const std::array<dealii::Tensor<1,dim,real>,nstate> &solution_gradient,
         const dealii::types::global_dof_index cell_index) const = 0;
 
-    /// Source terms additional to the baseline physics
+    /// Convective Numerical Split Flux for split form additional to the baseline physics (including Convective Numerical Split Flux for split form in additional PDEs of model)
+    virtual std::array<dealii::Tensor<1,dim,real>,nstate> 
+    convective_numerical_split_flux (
+        const std::array<real,nstate> &soln_const, 
+        const std::array<real,nstate> &soln_loop) const = 0;
+
+    /// Spectral radius of convective term Jacobian.
+    /** Used for scalar dissipation */
+    virtual std::array<real,nstate> convective_eigenvalues (
+        const std::array<real,nstate> &/*solution*/,
+        const dealii::Tensor<1,dim,real> &/*normal*/) const = 0;
+
+    /// Maximum convective eigenvalue used in Lax-Friedrichs
+    virtual real max_convective_eigenvalue (const std::array<real,nstate> &soln) const = 0;
+
+    /// Physical source terms additional to the baseline physics (including physical source terms in additional PDEs of model)
+    virtual std::array<real,nstate> physical_source_term (
+        const dealii::Point<dim,real> &pos,
+        const std::array<real,nstate> &solution,
+        const std::array<dealii::Tensor<1,dim,real>,nstate> &solution_gradient,
+        const dealii::types::global_dof_index cell_index) const;
+
+    /// Manufactured source terms additional to the baseline physics (including manufactured source terms in additional PDEs of model)
     virtual std::array<real,nstate> source_term (
         const dealii::Point<dim,real> &pos,
         const std::array<real,nstate> &solution,
         const dealii::types::global_dof_index cell_index) const = 0;
 
+    /// Evaluates boundary values and gradients on the other side of the face.
+    virtual void boundary_face_values (
+        const int /*boundary_type*/,
+        const dealii::Point<dim, real> &/*pos*/,
+        const dealii::Tensor<1,dim,real> &/*normal*/,
+        const std::array<real,nstate> &/*soln_int*/,
+        const std::array<dealii::Tensor<1,dim,real>,nstate> &/*soln_grad_int*/,
+        std::array<real,nstate> &/*soln_bc*/,
+        std::array<dealii::Tensor<1,dim,real>,nstate> &/*soln_grad_bc*/) const;
```

Comment:
make this a pure virtual

Reply:
see src/euler.cpp for the rans implementation, add std::abort for any boundary_type that is not implemented